### PR TITLE
Use async and await keywords in smoke tests

### PR DESCRIPTION
**Pull request by rodrigc**
## Contributor Checklist:

* [x] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

* Turn off Selenium promise manager.
* The Selenium promise manager is going away in Selenium 4.  We need to make our tests async as specified in this video:
    
    "Protractor: A New Hope"
    https://youtu.be/6aPfHrSl0Qk?t=1051

Presentation was given by @cnishina and @heathkit

### CODE REVIEW

**Comment by rodrigc** [comment]
@tardyp can you update metabotcfg with the fixes submitted in https://github.com/buildbot/metabbotcfg/pull/88 ?

**Comment by tardyp** [comment]
sure. I need to first wait for the build (didn't happen automatically for some reason)

**Comment by tardyp** [comment]
rebuilding

File: smokes/e2e/pages/builder.ts [hunk 1]
```diff
@@ -14,46 +15,50 @@ export class BuilderPage extends BasePage {
...
-        return element.all(By.buttonText(this.forceName)).first().click();
+    async goForce() {
+        await this.go();
+        await element.all(By.buttonText(this.forceName)).first().click();
     }
 
-    goBuild(buildRef) {
-        this.go();
-        return element.all(By.linkText(buildRef.toString())).click();
+    async goBuild(buildRef) {
```
Comment (cnishina) [inline smokes/e2e/pages/builder.ts]: More of a TypeScript nit, what is `buildRef` and why do you need to `toString()` it. Having types will help everyone using this method to make sure they use it properly.

Comment (cnishina) [reply smokes/e2e/pages/builder.ts]: Okay internal commentary: I understand this file was recently migrated over to TypeScript. So maybe the next PR?

File: smokes/e2e/hook.scenarios.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import {browser, by, element, ExpectedConditions as EC} from 'protractor';
```
Comment (cnishina) [inline smokes/e2e/hook.scenarios.ts]: Nit for prettiness. Missing a space to match other imports:
```
import { browser, by, element, ExpectedConditions as EC } from 'protractor';
```

File: smokes/e2e/pages/builder.ts [hunk 3]
```diff
@@ -14,46 +15,50 @@ export class BuilderPage extends BasePage {
...
-                return 0;
-            }
-            return elements[0].getText().then(numberstr => +numberstr);
-        });
+    async getLastSuccessBuildNumber() {
+        let elements = await element.all(By.css('span.badge-status.results_SUCCESS'));
+        if (elements.length === 0) {
+            return 0;
+        }
+        let numberstr = await elements[0].getText()
```
[the remark this answers is not in the document]
Comment (rodrigc) [reply smokes/e2e/pages/builder.ts]: Thanks for the review.  Fixed.

File: smokes/e2e/hook.scenarios.ts (hunk 2 above)
Comment (rodrigc) [reply smokes/e2e/hook.scenarios.ts]: Fixed

File: smokes/e2e/pages/builder.ts (hunk 1 above)
Comment (rodrigc) [reply smokes/e2e/pages/builder.ts]: Yes, we can clean this up in a subsequent PR.  This code was converted from CoffeeScript -> JavaScript -> TypeScript, so there are lots of subsequent small code cleanups that can be done.

Comment (cnishina) [reply smokes/e2e/pages/builder.ts]: Great! I'm excited to see what is next.

**Pull request merged**